roles/splunk/tasks/configure_splunk_secret.yml

```diff
@@ -7,4 +7,16 @@
     mode: 0644
   become: true
   notify: restart splunk
-  when: splunk_configure_secret
+  when: splunk_configure_secret and splunk_secret_file is defined
+
+- name: Install splunk.secret
```

Comment:
We should differentiate between the tasks in `name` if we're going to provide to multiple implementation methods. e.g. Line 1 "Install splunk.secret from file" and line 12 "Install splunk.secret from text"

Reply:
Added names matching your request so the tasks are different.